x-pack/filebeat/input/httpjson/config.go

```diff
@@ -84,6 +83,14 @@ func (c *config) Validate() error {
 			}
 		}
 	}
+	if c.OAuth2 != nil {
```

Comment:
```suggestion
	if c.OAuth2.IsEnabled() {
```

Reply:
Here I followed the same pattern as `Pagination`, which checks for it being enabled at the moment of using it, but still validates its config. To me it kind of made sense since even if disabled a user needs to introduce a valid config for it. Otherwise I would change also how `Pagination` is handled just to be consistent here and check for `IsEnabled` in both config validation and usage. WDYT?

Reply:
I think using `IsEnabled()` is correct now that you've refactored it a bit (before you we're correct). For example it would be invalid to use `api_key` or `authentication` only when oauth is enabled.

Reply:
Good point! :+1: